src/Uno.UWP/Media/Playback/MediaPlayer.wasm.cs

```diff
@@ -0,0 +1,297 @@
+			{
+				_isPlayRequested = false;
+				_isPlayerPrepared = false;
+			}
+		}
+
+		#region Player Initialization
+		private void InitializePlayer()
+		{
+			Player.OnSourceFailed -= OnError;
+			Player.OnSourceLoaded -= OnPrepared;
+			Player.OnSourceEnded -= OnCompletion;
+			Player.OnTimeUpdate -= OnTimeUpdate;
+			Player.OnSourceFailed += OnError;
+			Player.OnSourceLoaded += OnPrepared;
+			Player.OnSourceEnded += OnCompletion;
+			Player.OnTimeUpdate += OnTimeUpdate;
+
+			PlaybackSession.PlaybackStateChanged -= OnStatusChanged;
+			PlaybackSession.PlaybackStateChanged += OnStatusChanged;
+		}
+
+		private void SetPlaylistItems(MediaPlaybackList playlist)
+		{
+			_playlistItems = playlist.Items
+				.Select(i => i.Source.Uri)
+				.ToList();
+		}
+
+		protected virtual void InitializeSource()
+		{
+			PlaybackSession.NaturalDuration = TimeSpan.Zero;
+			PlaybackSession.PositionFromPlayer = TimeSpan.Zero;
+
+			// Reset player
+			TryDisposePlayer();
+			if (Source == null)
+			{
+				return;
+			}
+
+			try
+			{
+				InitializePlayer();
+				PlaybackSession.PlaybackState = MediaPlaybackState.Opening;
+
+				Uri uri;
+				switch (Source)
+				{
+					case MediaPlaybackList playlist when playlist.Items.Count > 0:
+						SetPlaylistItems(playlist);
+						uri = _playlistItems[0];
+						break;
+					case MediaPlaybackItem item:
+						uri = item.Source.Uri;
+						break;
+					case MediaSource source:
+						uri = source.Uri;
+						break;
+					default:
+						throw new InvalidOperationException("Unsupported media source type");
+				}
+
+				SetVideoSource(uri);
+				MediaOpened?.Invoke(this, null);
+			}
+			catch (global::System.Exception ex)
+			{
+				OnMediaFailed(ex);
+			}
+		}
+
+		private void OnTimeUpdate(object sender, object what)
+		{
+			PlaybackSession.PositionFromPlayer = Position;
+		}
+
+		private async void SetVideoSource(Uri uri)
+		{
+			if (!uri.IsAbsoluteUri || uri.Scheme == "")
+			{
+				uri = new Uri(MsAppXScheme + ":///" + uri.OriginalString.TrimStart("/"));
+			}
+
+			if (uri.IsLocalResource())
+			{
+				var file = await Windows.Storage.StorageFile.GetFileFromApplicationUriAsync(uri);
+				Player.Source = file.Path;
+				return;
+			}
+
+			if (uri.IsAppData())
+			{
+				var filePath = AppDataUriEvaluator.ToPath(uri);
+				Player.Source = filePath;
+				return;
+			}
+
+			Player.Source = uri.ToString();
+		}
+
+		#endregion
+
+		private void OnStatusChanged(MediaPlaybackSession sender, object args)
+		{
+			if ((MediaPlaybackState)args == MediaPlaybackState.Playing)
+			{
+				Player?.Play();
+			}
+		}
+
+		public virtual void Play()
+		{
+			if (Source == null || Player == null)
+			{
+				return;
+			}
+
+			try
+			{
+				// If we reached the end of media, we need to reset position to 0
+				if (PlaybackSession.PlaybackState == MediaPlaybackState.None)
+				{
+					PlaybackSession.Position = TimeSpan.Zero;
+				}
+
+				_isPlayRequested = true;
+
+				if (_isPlayerPrepared)
+				{
+					Player?.Play();
```

Comment:
Already checked for null above. If you worry it could have changed in between somehow, then store it in a local variable.
```suggestion
					Player.Play();
```